### PR TITLE
docs(sdk): Update sweep docstrings

### PR DESCRIPTION
**Pull request by ngrayluna**
Description
-----------
Provide more context for sweep docstrings. Word smithing as well. 

Testing
-------
How was this PR tested?

### CODE REVIEW

File: wandb/cli/cli.py [hunk 1]
```diff
@@ -824,12 +824,12 @@ def _clean():
...
-@click.option("--entity", "-e", default=None, help="The entity scope for the project.")
+@cli.command(context_settings=CONTEXT, help="Initialize a hyperparameter sweep. Search for hyperparameters that optimizes a cost function of a machine learning model by testing various combinations.")
+@click.option("--project", "-p", default=None, help="""The name of the project where W&B runs created from the sweep are sent to. If the project is not specified, the run is sent to a project labeled Uncategorized.""")
+@click.option("--entity", "-e", default=None, help="""The username or team name where you want to send W&B runs created by the sweep to. Ensure that the entity you specify already exists. If you don't specify an entity, the run will be sent to your default entity, which is usually your username.""")
 @click.option("--controller", is_flag=True, default=False, help="Run local controller")
 @click.option("--verbose", is_flag=True, default=False, help="Display verbose output")
-@click.option("--name", default=None, help="Set sweep name")
+@click.option("--name", default=None, help="The name of the sweep. The run ID that is created from the sweep is used if no name is specified.")
```
Comment (gtarpenning) [inline wandb/cli/cli.py:832]:
```suggestion
@click.option("--name", default=None, help="The name of the sweep. The sweep ID is used if no name is specified.")
```

**Review by gtarpenning — APPROVED** [review]
lgtm

File: wandb/cli/cli.py [hunk 2]
```diff
@@ -824,12 +824,12 @@ def _clean():
         _summary()
 
 
-@cli.command(context_settings=CONTEXT, help="Create a sweep")
-@click.option("--project", "-p", default=None, help="The project of the sweep.")
-@click.option("--entity", "-e", default=None, help="The entity scope for the project.")
+@cli.command(context_settings=CONTEXT, help="Initialize a hyperparameter sweep. Search for hyperparameters that optimizes a cost function of a machine learning model by testing various combinations.")
+@click.option("--project", "-p", default=None, help="""The name of the project where W&B runs created from the sweep are sent to. If the project is not specified, the run is sent to a project labeled Uncategorized.""")
```
Comment (kptkin) [inline wandb/cli/cli.py:828]: why do you need additional `""`?

File: wandb/sdk/wandb_sweep.py [hunk 3]
```diff
@@ -35,60 +35,31 @@ def sweep(
...
+        the run will be sent to your default entity,
+        which is usually your username.
+      project: The name of the project where W&B runs created from
+        the sweep are sent to. If the project is not specified, the
+        run is sent to a project labeled 'Uncategorized'.
 
     Returns:
       sweep_id: str. A unique identifier for the sweep.
-
-    Examples:
```
Comment (kptkin) [inline wandb/sdk/wandb_sweep.py:65]: why did you remove the example?

File: wandb/wandb_agent.py [hunk 4]
```diff
@@ -531,46 +531,33 @@ def run_agent(
...
-        project: (str, optional) W&B Project
-        count: (int, optional) the number of trials to run.
-
-    Examples:
-        Run a sample sweep over a function:
-        <!--yeadoc-test:one-parameter-sweep-agent-->
-        ```python
-        import wandb
-
-        sweep_configuration = {
```
Comment (kptkin) [inline wandb/wandb_agent.py:553]: again why did you remove the example?

**Review by kptkin — APPROVED** [review]

File: wandb/cli/cli.py (hunk 2 above)
Comment (ngrayluna) [reply wandb/cli/cli.py:828]: Good question, I added them because I figured that (eventually) we'd want to format it so it's not just one long line...

File: wandb/sdk/wandb_sweep.py (hunk 3 above)
Comment (ngrayluna) [reply wandb/sdk/wandb_sweep.py:65]: Ah. We (Docs Team) have an initiative to remove code examples and instead refer to the dev docs. 😬

**Comment by ngrayluna** [comment]
@kptkin When you get a chance, can you help unblock me and get this PR merged? 🙏

**Comment by kptkin** [comment]
@ngrayluna Few issues:

- This PR broke our ruff check, see here: https://app.circleci.com/pipelines/github/wandb/wandb/30179/workflows/0ffca0e4-b266-4ae6-a1af-dc3f575ba012/jobs/972005 (you can fix it by installing `tox` and running `tox -e ruff`, or just install the pre-commit hooks (see contributions docs)).
-  This PR also broke automated test for documentation when you removed these lines: https://github.com/wandb/wandb/pull/6830/files#diff-ba2b9dff0762c1dddea527141671b640b03ac23533713d1d19229c13e9691ef2L67-L91
see this error: https://app.circleci.com/pipelines/github/wandb/wandb/30179/workflows/0ffca0e4-b266-4ae6-a1af-dc3f575ba012/jobs/972040. To fix this you can either put this code back or just remove this file: tests/functional_tests/t0_main/doctest/one-parameter-sweep.yea (up to the sweeps team)

**Comment by ngrayluna** [comment]
@gtarpenning I removed this file t`ests/functional_tests/t0_main/doctest/one-parameter-sweep.yea` . Re: The Docs Team is removing code examples from the API Ref (for now).

**Review by gtarpenning — APPROVED** [review]
